src/lib/RichText/Renderer.php

```diff
@@ -452,30 +436,23 @@ protected function getEmbedTemplateName($resourceType, $isInline, $isDenied)
     /**
      * Check embed permissions for the given Content.
      *
-     * @throws \Symfony\Component\Security\Core\Exception\AccessDeniedException
-     *
-     * @param \Ibexa\Contracts\Core\Repository\Values\Content\Content $content
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
      */
     protected function checkContentPermissions(Content $content)
     {
         // Check both 'content/read' and 'content/view_embed'.
         if (
-            !$this->authorizationChecker->isGranted(
-                new AuthorizationAttribute('content', 'read', ['valueObject' => $content])
-            )
-            && !$this->authorizationChecker->isGranted(
-                new AuthorizationAttribute('content', 'view_embed', ['valueObject' => $content])
-            )
+            !$this->permissionResolver->canUser('content', 'read', $content)
+            && !$this->permissionResolver->canUser('content', 'view_embed', $content) // Shouldn't we have `||` here?
```

Comment:
Having either `content/read` or `content/view_embed` instead of both of them seems to me reasonable enough.